defender/downloaded/Datadog_Alerts_Heat_Map.js

```diff
@@ -0,0 +1,267 @@
+    operationName: 'Retrieve',
+    query: `query Retrieve($getListInput: GetAlertsInput) {
+      getList(input: $getListInput) {
+        alerts {
+          hash
+          description
+          severity
+          protocol
+          name
+          everest_id
+          alert_id
+          scanner_count
+          source {
+            tx_hash
+            agent {
+              id
+              name
+            }
+            block {
+              chain_id
+              number
+              timestamp
+            }
+          }
+          projects {
+            id
+            name
+          }
+        }
+        nextPageValues {
+          blocknumber
+          id
+        }
+        currentPageValues {
+          blocknumber
+          id
+        }
+      }
+    }`,
+    variables: {
+      getListInput: {
+        severity: [],
+        startDate: (lastUpdateTimestamp + 1).toString(),
+        endDate: currentTimestamp.toString(),
+        txHash: '',
+        text: '',
+        muted: [],
+        sort: 'desc',
+        agents: [botId],
+        addresses: [],
+        project: '',
+      },
+    },
+  };
+  return graphqlQuery;
+}
+
+async function postQuery(graphqlQuery) {
+  const headers = {
+    'content-type': 'application/json',
+  };
+
+  // perform the POST request
+  const response = await axios({
+    url: fortaExplorerApiEndpoint,
+    method: 'post',
+    headers,
+    data: graphqlQuery,
+  });
+
+  return response;
+}
+
+async function postToDatadog(data, apiKey, url) {
+  const headers = {
+    'Content-Type': 'application/json',
+    'DD-API-KEY': apiKey,
+  };
+
+  // perform the POST request
+  const response = await axios({
+    url,
+    method: 'post',
+    headers,
+    data,
+  });
+
+  return response;
+}
+
+// entry point for autotask
+// eslint-disable-next-line func-names
+exports.handler = async function (autotaskEvent) {
+  // get the current timestamp once
+  // this value will be used across all queries to determine how much data to
+  // retrieve
+  const currentTimestamp = (new Date()).getTime();
+  console.debug(`currentTimestamp: ${currentTimestamp.toString()}`);
+
+  console.debug(JSON.stringify(autotaskEvent, null, 2));
+
+  const { secrets } = autotaskEvent;
+  if (secrets === undefined) {
+    throw new Error('secrets undefined');
+  }
+
+  const { DatadogApiKey: datadogApiKey } = secrets;
+  if (datadogApiKey === undefined) {
+    throw new Error('Datadog API key undefined');
+  }
+
+  let firstRun = false;
+  const store = new KeyValueStoreClient(autotaskEvent);
+
+  // load the latest timestamp that was stored
+  let lastUpdateTimestamp = await store.get('lastUpdateTimestampAlerts');
+
+  // the first time this Autotask is executed, we will need to manually set the value
+  // of the last timestamp
+  if (lastUpdateTimestamp === undefined || lastUpdateTimestamp === null) {
+    console.debug('Autotask run for the first time, initializing lastUpdateTimestamp');
+    firstRun = true;
```

Comment:
```suggestion
```
`firstRun` is assigned a value but never used

Reply:
removed